### PR TITLE
remain toc relative path without ~/ if unable to resolve

### PR DESCRIPTION
**Pull request by superyyrrzz**
@chenkennt @fenxu @hellosnow @ansyral @qinezh @vwxyzh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs [hunk 1]
```diff
@@ -165,34 +169,34 @@ private void RegisterTocMap(TocItemViewModel item, string key, IDocumentBuildCon
...
-                return originalPathToFile;
+                return pathToFile;
             }
 
-            var index = originalPathToFile.IndexOf('#');
+            var index = pathToFile.IndexOf('#');
             if (index == 0)
             {
-                throw new DocumentException($"Invalid toc link: {originalPathToFile}.");
+                throw new DocumentException($"Invalid toc link: {pathToFile}.");
```
Comment (fenxu) [inline src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs]: >pathToFile [](start = 65, length = 10)

Shall we use originalPathToFile value to tell give user the error?

File: src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs [hunk 2]
```diff
@@ -165,34 +169,34 @@ private void RegisterTocMap(TocItemViewModel item, string key, IDocumentBuildCon
...
-                ? context.GetFilePath(originalPathToFile)
-                : context.GetFilePath(originalPathToFile.Remove(index));
+                ? context.GetFilePath(pathToFile)
+                : context.GetFilePath(pathToFile.Remove(index));
 
 
             if (href == null)
             {
-                Logger.LogWarning($"Unable to find file \"{originalPathToFile}\" referenced by TOC file \"{model.LocalPathFromRepoRoot}\"");
+                Logger.LogWarning($"Unable to find file \"{pathToFile}\" referenced by TOC file \"{model.LocalPathFromRepoRoot}\"");
```
Comment (fenxu) [inline src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs]: >pathToFile [](start = 59, length = 10)

So does this, then user could see the original content he write in the toc file. But not see ~/ any more.

**Comment by fenxu** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs (hunk 1 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs]: OK. seems relative path is clearer.

---
In reply to: [75433940](https://github.com/dotnet/docfx/pull/623#discussion_r75433940) [](ancestors = 75433940)

**Pull request merged**